run-checks

```diff
@@ -158,15 +158,18 @@ if [ "$STATIC" = 1 ]; then
     if [ -z "${SKIP_GOFMT:-}" ]; then
         echo Checking formatting
         fmt=""
-        for dir in $(go list -f '{{.Dir}}' ./... | grep -v '/vendor/' ); do
-            s="$(${GOFMT:-gofmt} -s -l -d "$dir" | grep -v /vendor/ || true)"
+        for dir in $(go list -f '{{.Dir}}' ./... | grep -v '/vendor/' | grep -E 'snapd/[A-Za-z0-9_]+$' ); do
+            # skip vendor packages
+            # skip subpackages of packages under snapd, gofmt already inspects them
+            s="$(${GOFMT:-gofmt} -s -l -d "$dir" || true)"
             if [ -n "$s" ]; then
                 fmt="$s\\n$fmt"
             fi
         done
         if [ -n "$fmt" ]; then
             echo "Formatting wrong in following files:"
             echo "$fmt" | sed -e 's/\\n/\n/g'
+            exit 1
```

Comment:
Wil this cause pain for the people already on newer go version than 1.10 (like most of us :) ?

Reply:
Hm, it got dropped with the refactor introduced in e9ba5013a6. @zyga do you remember whether the refactor was supposed to disable exit 1 or just change how SKIP_GOFMT is handled?

Reply:
Hmm... I only recall where all of the checks were disabled when SKIP_GOFMT was enabled.

Reply:
Ok, so perhaps removal of `exit 1` in #8186 was by accident then.

Reply:
Oh, indeed. This is protected by SKIP_GOFMT so it looks like it should do the right thing (exit 1) if not given.

Reply:
good catch here!